cpp/src/structure/select_random_vertices_impl.hpp

```diff
@@ -105,11 +168,32 @@ rmm::device_uvector<vertex_t> select_random_vertices(
 
       std::vector<size_t> tx_value_counts(comm_size);
       for (int i = 0; i < comm_size; ++i) {
-        tx_value_counts[i] =
-          mg_sample_buffer.size() / comm_size +
-          (static_cast<size_t>(i) < static_cast<size_t>(mg_sample_buffer.size() % comm_size) ? 1
-                                                                                             : 0);
+        tx_value_counts[i] = mg_sample_buffer.size() / comm_size;
+      }
```

Comment:
This can be `std::fill(tx_value_counts.begin(), tx_value_counts.end(), mg_sample_buffer.size()  /comm_size);`